pkg/cmd/server/server.go

```diff
@@ -706,12 +711,24 @@ func (s *server) runControllers(config *api.Config, defaultBackupLocation *api.B
 		s.config.restoreResourcePriorities,
 		s.arkClient.ArkV1(),
 		s.kubeClient.CoreV1().Namespaces(),
+		s.kubeSharedInformerFactory.Core().V1().Namespaces(),
+		s.kubeClient.CoreV1().PersistentVolumes(),
+		s.kubeSharedInformerFactory.Core().V1().PersistentVolumes(),
 		s.resticManager,
 		s.config.podVolumeOperationTimeout,
+		s.config.namespaceTimeout,
 		s.logger,
 	)
 	cmd.CheckError(err)
 
+	// Start the informer and wait for cache sync here so that the event handlers added in NewKubernetesRestorer are ready by the time
+	// the restore controller is started
+	s.logger.Debugln("Starting kube shared informer factory")
+	go s.kubeSharedInformerFactory.Start(ctx.Done())
+	s.logger.Debugln("Waiting for kubesharedinformer to sync")
+	cache.WaitForCacheSync(ctx.Done(), s.kubeSharedInformerFactory.Core().V1().Namespaces().Informer().HasSynced)
```

Comment:
check return val here

Reply:
I notice we don't check this elsewhere in the server - should I add that? Is it a total failure mode if it returns `false`?